onnxruntime/core/framework/execution_frame.cc

```diff
@@ -387,6 +395,15 @@ Status ExecutionFrame::AllocateMLValueTensorSelfOwnBufferHelper(OrtValue& ort_va
     TraceAllocate(ort_value_index, size);
   }
 
+  
+  {
+    // This code block is not thread-safe.
+    // Dynamic activation size would be accessed by multiple threads
+    // if parallel executor is used.
+    std::unique_lock<std::mutex> lock(mtx_);
```

Comment:
> std::unique_lock<std::mutex> lock(mtx_); [](start = 3, length = 41)

mutex should be shared between read and write

Reply:
Not sure what should I do. Do you mean a ThreadSafeUnorderedMap so that those accesses are always safe?

---
In reply to: [466617105](https://github.com/microsoft/onnxruntime/pull/4712#discussion_r466617105) [](ancestors = 466617105)